app/main.py

```diff
@@ -35,3 +37,47 @@ async def all_occurence(request: Request, term: str):
     parser = CGUsAllOccurencesParser(Path(CGUS_DATASET_PATH), term)
     parser.run()
     return parser.to_dict()
+
+@app.get(f"{BASE_PATH}/list_services/v1/")
+@limiter.limit(RATE_LIMIT)
+async def list_services(request: Request):
+    """
+    Returns a JSON object with services as keys and a list of their available document types.
+    """
+    dataset = CGUsDataset(Path(CGUS_DATASET_PATH))
+    return dataset.list_all_services_doc_types()
+    
+@app.get(f"{BASE_PATH}/get_snapshot_at_date/v1/{{service}}/{{document_type}}/{{date}}")
+@limiter.limit(RATE_LIMIT)
+async def get_snapshot_at_date(request: Request, service: str, document_type: str, date: str):
```

Comment:
`get_snapshot_at_date` should named `get_version_at_date` because this API route return version data, not snapshot.